File: keemsebin/Greedy/BOJ_1541.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("-");

let ans = input.map((num) => {
  let sum = 0;
  // +를 기준으로 나누어서 각 요소를 더해준다.
  num.split("+").forEach((item) => {
    sum += parseInt(item);
  });
  return sum;
});
// 첫번째 요소를 제외한 나머지 요소를 빼준다.

console.log(ans.reduce((acc, cur) => acc - cur));

File: comgongdaeng/Greedy/BOJ_1541.py
import sys
# - 뒤에 괄호를 치면 최솟값을 얻을 수 있음.
# 숫자만, 혹은 덧셈으로만 이루어진 문자열로 구성된 exp
exp = sys.stdin.readline().rstrip().split('-')

# exp[0]에서 exp[1], exp[2]... 를 전부 빼주면 됨
ans = sum(map(int, (exp[0].split('+'))))

for i in exp[1:]:
    i = sum(map(int, (i.split('+'))))
    ans -= i

print(ans)